### PR TITLE
 Implement identification primitives

### PR DESCRIPTION
**Pull request by belochub**
Refs: https://github.com/metarhia/globalstorage/issues/8
Refs: https://github.com/metarhia/Metarhia/issues/21

### CODE REVIEW

File: lib/identification.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+class RemoteServer extends StorageProvider {
+  constructor({
+    host,
+    ports,
+  }, providerOptions) {
+    super(providerOptions);
+    this.host = host;
+    this.ports = ports;
+    if (this.ports.length > 1) {
+      this.ports = sequence(this.ports, 0);
```
Comment (tshemsedinov) [inline lib/identification.js]: `this.ports = ports.length > 1 ? sequence(this.ports, 0) : ports`

File: lib/identification.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      getTreeDepth(node[1], depth + 1)
+    );
+  }
+  return depth;
+};
+
+const arrayifyTree = node => {
+  if (node[0] && node[1]) {
+    return [arrayifyTree(node[0]), arrayifyTree(node[1])];
+  } else {
```
Comment (tshemsedinov) [inline lib/identification.js]:
```suggestion
  }
```

File: lib/identification.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    return node;
```
Comment (tshemsedinov) [inline lib/identification.js]:
```suggestion
  return node;
```

File: lib/identification.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  }
```
Comment (tshemsedinov) [inline lib/identification.js]:
```suggestion
```

Comment (tshemsedinov) [inline lib/identification.js]:
```js
  }
  return node;
}
```

File: test/identification.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        host: 'gs.metarhia.com',
+        ports: [1234],
+      },
+    ],
+  ],
+};
+
+const contains = (test, object, expected) => {
+  for (const key in expected) {
+    if (!expected.hasOwnProperty(key)) {
```
Comment (lundibundi) [inline test/identification.js]: I'd prefer `Object.keys` instead of manual checking properties here.

**Comment by belochub** [comment]
Ping @nechaido, @tshemsedinov.

**Comment by lundibundi** [comment]
ping @tshemsedinov

**Pull request merged**